product-service/src/functions/getProductsById/handler.ts

```diff
@@ -0,0 +1,26 @@
+import { APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import { middyfy } from '@libs/lambda';
+import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { notFoundResponse, okResponse } from '../../responses';
+import { databaseService } from 'src/dependencies';
+import { IDatabaseService } from 'src/services/database-service';
+
+const getProductsById = (databaseService: IDatabaseService): ValidatedEventAPIGatewayProxyEvent<APIGatewayProxyResult> => async (event) => {
+  const productId = event?.pathParameters?.id;
```

Comment:
What if you get a DB error in lambda? In this case, lambda can retry a few times, which we don't want. You would need to put everything here in try-catch block and in case of an error return with an error response and log error to the console (not return it to the client)